libraries/Microsoft.Bot.Builder.Skills/ChannelApiMiddleware.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Skills
+{
+    /// <summary>
+    /// Handles InvokeActivity for ChannelAPI methods calls coming from SkillHostController.
+    /// </summary>
+    internal class ChannelApiMiddleware : IMiddleware
+    {
+        private readonly SkillAdapter _skillAdapter;
+
+        internal ChannelApiMiddleware(SkillAdapter skillAdapter)
+        {
+            _skillAdapter = skillAdapter;
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default)
+        {
+            // register the skill adapter so people can get it to do .ForwardActivityAsync()
+            turnContext.TurnState.Add(_skillAdapter);
```

Comment:
>turnContext.TurnState.Add(_skillAdapter); [](start = 12, length = 41)

why do we need to do this?  The skill adapter can just be passed to the constructor of a skill dialog.